src/pages/signin/ChooseSSOOrMagicCode.tsx

```diff
@@ -10,44 +10,40 @@ import useLocalize from '@hooks/useLocalize';
 import useNetwork from '@hooks/useNetwork';
 import useResponsiveLayout from '@hooks/useResponsiveLayout';
 import useThemeStyles from '@hooks/useThemeStyles';
-import * as ErrorUtils from '@libs/ErrorUtils';
+import {getLatestErrorMessage} from '@libs/ErrorUtils';
 import Navigation from '@libs/Navigation/Navigation';
-import * as Session from '@userActions/Session';
+import {clearSignInData, resendValidateCode} from '@userActions/Session';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {Account, Credentials} from '@src/types/onyx';
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import ChangeExpensifyLoginLink from './ChangeExpensifyLoginLink';
 import Terms from './Terms';
 
-type ChooseSSOOrMagicCodeOnyxProps = {
-    /** The credentials of the logged in person */
-    credentials: OnyxEntry<Credentials>;
-
-    /** The details about the account that the user is signing in with */
-    account: OnyxEntry<Account>;
-};
-
-type ChooseSSOOrMagicCodeProps = ChooseSSOOrMagicCodeOnyxProps & {
+type ChooseSSOOrMagicCodeProps = {
     /** Function that returns whether the user is using SAML or magic codes to log in */
     setIsUsingMagicCode: (value: boolean) => void;
 };
 
-function ChooseSSOOrMagicCode({credentials, account, setIsUsingMagicCode}: ChooseSSOOrMagicCodeProps) {
+function ChooseSSOOrMagicCode({setIsUsingMagicCode}: ChooseSSOOrMagicCodeProps) {
     const styles = useThemeStyles();
     const {isKeyboardShown} = useKeyboardState();
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
     const {shouldUseNarrowLayout} = useResponsiveLayout();
 
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT, {canBeMissing: false});
+    const [credentials] = useOnyx(ONYXKEYS.CREDENTIALS, {canBeMissing: true});
+
     // This view doesn't have a field for user input, so dismiss the device keyboard if shown
-    useEffect(() => {
-        if (!isKeyboardShown) {
-            return;
-        }
-        Keyboard.dismiss();
-    }, [isKeyboardShown]);
+    useFocusEffect(
+        useCallback(() => {
+            if (!isKeyboardShown) {
+                return;
+            }
+            Keyboard.dismiss();
+        }, [isKeyboardShown]),
+    );
```

Comment:
Does anybody remember why we added this change?

Reply:
Ok, it has been added [here](https://github.com/software-mansion-labs/expensify-app-fork/commit/e23956a0c3eb39523a28a7cfc2eaffd709fcbbad) by me 💀 Copletely forgot about that change 😓 

It's probably related to the keyboard showing on wrong screen when logging with saml